cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -149,6 +150,40 @@ inline sycl::event copy(sycl::queue& queue,
     return memcpy(queue, dest, src, sizeof(T) * n, deps);
 }
 
+template <typename T>
+inline sycl::event copy_host2usm(sycl::queue& queue,
+                                 T* dest_usm,
+                                 const T* src_host,
+                                 std::int64_t count,
+                                 const event_vector& deps = {}) {
+    ONEDAL_ASSERT(count > 0);
+    ONEDAL_ASSERT(is_known_usm(queue, dest_usm));
+
+    // TODO: Remove additional copy to host usm memory once
+    //       bug in `copy` with the host memory is fixed
+    T* tmp_usm_host = malloc_host<T>(queue, count);
```

Comment:
What about memory deallocation? Tip: use `unique_usm_ptr` defined below

Reply:
Obvious memory leak 😢 
thank you!